modules/extensions-support/src/main/java/org/mule/module/extension/internal/model/property/SubtypesModelProperty.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.model.property;
+
+
+import org.mule.extension.api.introspection.ModelProperty;
+import org.mule.module.extension.internal.util.IntrospectionUtils;
+import org.mule.util.CollectionUtils;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * An immutable model property which specifies the declared implementations
+ * for the {@link ModelProperty}'s holder type
+ *
+ * @since 4.0
+ */
+public final class SubtypesModelProperty implements ModelProperty
+{
+
+    private final List<Class<?>> subtypes;
```

Comment:
I would keep a list of MetadataTypes.. otherwise, how can the tooling generate editors for it